provisioner/shell-local/communicator.go

```diff
@@ -0,0 +1,81 @@
+package shell
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"syscall"
+
+	"github.com/mitchellh/packer/packer"
+	"github.com/mitchellh/packer/template/interpolate"
+)
+
+type Communicator struct {
+	ExecuteCommand []string
+	Ctx            interpolate.Context
+}
+
+func (c *Communicator) Start(cmd *packer.RemoteCmd) error {
+	// Render the template so that we know how to execute the command
+	c.Ctx.Data = &ExecuteCommandTemplate{
+		Command: cmd.Command,
+	}
+	for i, field := range c.ExecuteCommand {
+		command, err := interpolate.Render(field, &c.Ctx)
+		if err != nil {
+			return fmt.Errorf("Error processing command: %s", err)
+		}
+
+		c.ExecuteCommand[i] = command
+	}
+
+	// Build the local command to execute
+	localCmd := exec.Command(c.ExecuteCommand[0], c.ExecuteCommand[1:]...)
+	localCmd.Stdin = cmd.Stdin
+	localCmd.Stdout = cmd.Stdout
+	localCmd.Stderr = cmd.Stderr
+
+	// Start it. If it doesn't work, then error right away.
+	if err := localCmd.Start(); err != nil {
+		return err
+	}
+
+	// We've started successfully. Start a goroutine to wait for
+	// it to complete and track exit status.
+	go func() {
+		var exitStatus int
+		err := localCmd.Wait()
```

Comment:
Is there a channel or wait group wrapped around `Communicator.Start()`?

Since this wait is called in a goroutine it could be scheduled after the function returns. If it returns and exits the local command will never run.

I noticed the test *also* calls `cmd.Wait()` but this control flow seems awkward. Maybe Start() should return a channel like iochan does?